core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -349,34 +349,7 @@ class DAGScheduler(
   }
 
   private def getMissingParentStages(stage: Stage): List[Stage] = {
-    val missing = new HashSet[Stage]
-    val visited = new HashSet[RDD[_]]
-    // We are manually maintaining a stack here to prevent StackOverflowError
-    // caused by recursively visiting
-    val waitingForVisit = new Stack[RDD[_]]
-    def visit(rdd: RDD[_]) {
-      if (!visited(rdd)) {
-        visited += rdd
-        if (getCacheLocs(rdd).contains(Nil)) {
```

Comment:
Actually I don't fully understand why we need this. Does this mean the `rdd` has at least one block which hasn't been cached? And why we need this to get missing parent stages?
@lianhuiwang @rxin